packages/components/data-table/src/data-table.tsx

```diff
@@ -21,11 +21,17 @@ import DataRow from './data-row';
 import useManualColumnResizing from './use-manual-column-resizing-reducer';
 import ColumnResizingContext from './column-resizing-context';
 
+export type TRow = {
+  id: string;
+  [key: string]: string;
+};
```

Comment:
We don't nee this anymore. The generic signature of `DataTable` should be something like

```ts
interface TRow {
  id: string
}

const DataTable = <Row extends TRow = TRow>(props: TProps<Row>)
```

All types accepting the generic argument should also implement the same signature. For example:

```ts
type TProps<Row extends TRow = TRow> = {
  // ...
}
```